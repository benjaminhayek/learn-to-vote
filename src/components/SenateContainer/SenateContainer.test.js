import React from 'react';
import { SenateContainer } from './index';
import { shallow } from 'enzyme';

describe('SenateContainer', () => {
  let wrapper;
  let senator;
  
  beforeEach(() => {
    senator = [{name: 'Bob', id: 2, party: 'R', title: 'Senator', nextElection: '2020', selected: false}, {name: 'Jan', id: 2, party: 'R', title: 'Senator', nextElection: '2018', selected: false}];    
    wrapper = shallow(<SenateContainer senators={senator}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should have an initial state of bills set to an empty array', () => {
    expect(wrapper.state().bills).toEqual([]);
  });

  it('should set the state of bills when handleSubmit is envoked', () => {
    const expected = []
    wrapper.instance().handleSubmit();
    expect(wrapper.state().bills).toEqual(expected);
  });
})