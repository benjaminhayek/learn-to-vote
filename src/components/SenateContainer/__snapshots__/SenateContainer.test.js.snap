// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SenateContainer should match the snapshot 1`] = `
<div>
  <div>
    <h1
      className="container-title"
    >
      Your Senators
    </h1>
    <h1
      className="card-container"
    >
      <Connect(SenateCard)
        key="b1d0c4b4-744f-414b-b266-f61bbaec1a2b"
        senator={
          Object {
            "id": 2,
            "name": "Bob",
            "nextElection": "2020",
            "party": "R",
            "selected": false,
            "title": "Senator",
          }
        }
      />
      <Connect(SenateCard)
        key="d0e22592-6c09-4849-9723-70e8fd4f76e6"
        senator={
          Object {
            "id": 2,
            "name": "Jan",
            "nextElection": "2018",
            "party": "R",
            "selected": false,
            "title": "Senator",
          }
        }
      />
    </h1>
  </div>
</div>
`;
